log/cache.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"bytes"
+
+	"github.com/google/trillian/merkle/compact"
+	"github.com/google/trillian/types"
+)
+
+// TreeCache caches per-tree information between sequencing runs.
+//
+// TODO(pavelkalinnikov): Support some cache clearing mechanism.
+type TreeCache struct {
+	ct map[int64]*compact.Tree
```

Comment:
You may want to use `sync.Map` here, or otherwise protect with `sync.Mutex`?

Reply:
Fantastic point.